sympy/printing/tests/test_maple.py

```diff
@@ -0,0 +1,413 @@
+    assert maple_code(A) == "Matrix([1,sin(2/x),3*pi/(5*x)], storage = triangle)"
+    assert maple_code(A.T) == "[1, sin(2/x), 3*pi/(5*x)]"
+
+
+@XFAIL
+def test_Matrices_entries_not_hadamard():
+    # For Matrix with col >= 2, row >= 2, they need to be scalars
+    # FIXME: is it worth worrying about this?  Its not wrong, just
+    # leave it user's responsibility to put scalar data for x.
+    A = Matrix([[1, sin(2 / x), 3 * pi / x / 5], [1, 2, x * y]])
+    expected = ("[1 sin(2/x) 3*pi/(5*x);\n"
+                "1        2        x*y]")  # <- we give x*y
+    assert maple_code(A) == expected
+
+
+def test_MatrixSymbol():
+    n = Symbol('n', integer=True)
+    A = MatrixSymbol('A', n, n)
+    B = MatrixSymbol('B', n, n)
+    assert maple_code(A * B) == "A.B"
+    assert maple_code(B * A) == "B.A"
+    assert maple_code(2 * A * B) == "2*A.B"
+    assert maple_code(B * 2 * A) == "2*B.A"
+
+    # FIXME: need more info about eye.
+    assert maple_code(A * (B + 3 * Identity(n))) == "A*(3*eye(n) + B)"
+    assert maple_code(A ** (x ** 2)) == "A^(x^2)"
+    assert maple_code(A ** 3) == "A^3"
+    assert maple_code(A ** (S.Half)) == "A^(1/2)"
+
+
+def test_special_matrices():
+    # FIXME: need more info about eye.
+    assert maple_code(6 * Identity(3)) == "6*eye(3)"
+
+
+def test_containers():
+    # FIXME: need more info.
+    assert maple_code([1, 2, 3, [4, 5, [6, 7]], 8, [9, 10], 11]) == \
+           "Any[1, 2, 3, Any[4, 5, Any[6, 7]], 8, Any[9, 10], 11]"
+    assert maple_code((1, 2, (3, 4))) == "(1, 2, (3, 4))"
+    assert maple_code([1]) == "Any[1]"
+    assert maple_code((1,)) == "(1,)"
+    assert maple_code(Tuple(*[1, 2, 3])) == "(1, 2, 3)"
+    assert maple_code((1, x * y, (3, x ** 2))) == "(1, x*y, (3, x.^2))"
+    # scalar, matrix, empty matrix and empty list
+    assert maple_code((1, eye(3), Matrix(0, 0, []), [])) == "(1, [1 0 0;\n0 1 0;\n0 0 1], zeros(0, 0), Any[])"
+
+
+def test_julia_noninline():
+    # FIXME: need more info.
+    source = maple_code((x + y) / Catalan, assign_to='me', inline=False)
+    expected = (
+                   "const Catalan = %s\n"
+                   "me = (x + y)/Catalan"
+               ) % Catalan.evalf(17)
+    assert source == expected
+
+
+"""
+ _____ _                     _       _         _            
+|_   _| |_  ___   __ ___  __| |___  | |__  ___| |_____ __ __
+  | | | ' \/ -_) / _/ _ \/ _` / -_) | '_ \/ -_) / _ \ V  V /
+  |_| |_||_\___| \__\___/\__,_\___| |_.__/\___|_\___/\_/\_/ 
+                                                            
+                 _   _         _                         _ _  __ _        _ 
+ _ _  ___ ___ __| | | |_ ___  | |__  ___   _ __  ___  __| (_)/ _(_)___ __| |
+| ' \/ -_) -_) _` | |  _/ _ \ | '_ \/ -_) | '  \/ _ \/ _` | |  _| / -_) _` |
+|_||_\___\___\__,_|  \__\___/ |_.__/\___| |_|_|_\___/\__,_|_|_| |_\___\__,_|
+                                                                            
+"""
+
+
+def test_julia_piecewise():
```

Comment:
It would be better to write piecewise directly to other CAS implementation, if they already have their own piecewise implementation

Reply:
Thanks.